### PR TITLE
Add error messages to 404's caused by cors closes #2857

### LINKED ISSUE #2857: Add CORS errors to 404 responses
Should we add the reason why the CORS preflight request failed to the 404 payload?

### PR DESCRIPTION
**Pull request by AdriVanHoudt**
Closes #2857

### CODE REVIEW

**Comment by AdriVanHoudt** [comment]
Btw any comments about the messages welcome, just put ones there that made sense to me.

File: lib/cors.js [hunk 1]
```diff
@@ -87,30 +87,30 @@ internals.handler = function (request, reply) {
...
-        return reply(Boom.notFound());
+        return reply(Boom.notFound('Missing access-control-request-method header'));
     }
 
     // Lookup route
 
     var route = request.connection.match(method, request.path, request.headers.host);
     if (!route) {
-        return reply(Boom.notFound());
+        return reply(Boom.notFound('Missing route'));
```
Comment (hueniverse) [inline lib/cors.js]: This is a classic not found. No need for a message.

File: lib/cors.js [hunk 2]
```diff
@@ -87,30 +87,30 @@ internals.handler = function (request, reply) {
...
     var origin = request.headers.origin;
     if (!origin) {
-        return reply(Boom.notFound());
+        return reply(Boom.notFound('Missing origin header'));
     }
 
     var method = request.headers['access-control-request-method'];
     if (!method) {
-        return reply(Boom.notFound());
+        return reply(Boom.notFound('Missing access-control-request-method header'));
```
Comment (hueniverse) [inline lib/cors.js]: Capitalize the header words

File: lib/cors.js [hunk 3]
```diff
@@ -87,30 +87,30 @@ internals.handler = function (request, reply) {
 
     var origin = request.headers.origin;
     if (!origin) {
-        return reply(Boom.notFound());
+        return reply(Boom.notFound('Missing origin header'));
```
Comment (hueniverse) [inline lib/cors.js]: Origin

File: lib/cors.js [hunk 4]
```diff
@@ -87,30 +87,30 @@ internals.handler = function (request, reply) {
...
     if (!settings) {
-        return reply(Boom.notFound());
+        return reply(Boom.notFound('CORS is disabled for this route'));
     }
 
     // Validate Origin header
 
     if (!internals.matchOrigin(origin, settings)) {
-        return reply(Boom.notFound());
+        return reply(Boom.notFound('Origin header mismatch'));
```
Comment (hueniverse) [inline lib/cors.js]: Origin not allowed

**Comment by AdriVanHoudt** [comment]
Updated the messages

**Pull request merged**